shop/src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -0,0 +1,108 @@
+package ro.msg.learning.shop.service;
+
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.*;
+import ro.msg.learning.shop.domain.key.OrderDetailId;
+import ro.msg.learning.shop.domain.key.StockId;
+import ro.msg.learning.shop.dto.CreateOrderDto;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.OrderProductDto;
+import ro.msg.learning.shop.exception.BadRequestException;
+import ro.msg.learning.shop.mapper.OrderDetailMapper;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.mapper.StockMapper;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.strategy.OrderLocationStrategy;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.IntStream;
+
+import static ro.msg.learning.shop.message.Messages.*;
+import static ro.msg.learning.shop.message.Messages.BAD_PRODUCT;
+
+@Service
+@AllArgsConstructor
+public class OrderService {
+    private OrderRepository orderRepository;
+    private CustomerRepository customerRepository;
+    private StockRepository stockRepository;
+    private ProductRepository productRepository;
+    private OrderDetailMapper orderDetailMapper;
+    private OrderLocationStrategy orderLocationStrategy;
+    private OrderMapper orderMapper;
+    private StockMapper stockMapper;
+
+    public OrderDto create(CreateOrderDto createOrderDto) {
+        if (customerRepository.findById(UUID.fromString(createOrderDto.getCustomerId())).isEmpty()) {
+            throw new BadRequestException(BAD_USER);
+        }
+
+        if (productRepository.findById(UUID.fromString(createOrderDto.getOrderProductDtoList().get(0).getProductId())).isEmpty()) {
+            throw new BadRequestException(BAD_PRODUCT);
+        }
+
+        for (int i = 0; i < createOrderDto.getOrderProductDtoList().size(); i++) {
```

Comment:
You have this for statement and also the one after that iterate through the same things. You can keep only one and also use that format that we've talked about with for `(OrderProductDto orderProductDto : orderProductDtos)`